query-dsl/src/main/java/org/infinispan/query/dsl/Query.java

```diff
@@ -88,6 +87,21 @@
 
    Query<T> maxResults(int maxResults);
 
+   /**
+    * @return current hitCountAccuracy if present
+    * @see #hitCountAccuracy(int)
+    */
+   Optional<Integer> hitCountAccuracy();
+
+   /**
+    * Limit the required accuracy of the hit count for the indexed queries to an upper-bound.
+    * Setting the hit-count-accuracy could improve the performance of queries targeting large data sets.
+    *
+    * @param hitCountAccuracy The value to apply
+    * @return <code>this</code>, for method chaining
+    */
+   Query<T> hitCountAccuracy(int hitCountAccuracy);
```

Comment:
Should we have a way to unset the value as well? I noticed it is an Optional return value above.

Reply:
I think that this should be consistent with the other mutable properties (e.g., startOffset and maxResults), and they do not have unset methods.